groups/infrastructure/route-53.tf

```diff
@@ -0,0 +1,21 @@
+resource "aws_acm_certificate" "certificate" {
+  domain_name       = local.domain_name
+  validation_method = "DNS"
+
+  tags = {
+    Environment = var.environment
+  }
+}
+
+resource "aws_route53_record" "route" {
+  name    = aws_acm_certificate.certificate.domain_validation_options[0].resource_record_name
+  type    = aws_acm_certificate.certificate.domain_validation_options[0].resource_record_type
+  zone_id = data.aws_route53_zone.hosted_zone.id
+  records = [aws_acm_certificate.certificate.domain_validation_options[0].resource_record_value]
+  ttl     = 60
+}
+
+resource "aws_acm_certificate_validation" "certificate" {
+  certificate_arn         = aws_acm_certificate.certificate.arn
+  validation_record_fqdns = [aws_route53_record.certificate_validation.fqdn]
+}
```

Comment:
You're referencing a resource `aws_route53_record` with label `certificate_validation` but one doesn't exist. You have one with a label of `route` and so either the label on the `aws_route53_record` resource should be updated accordingly, or the reference updated to `aws_route53_record.route.fqdn`